compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -1515,7 +1515,12 @@ LogicalResult TensorBitCastOp::verify() {
                                  {getResultDims()}))) {
     return failure();
   }
-
+  auto sourceType = cast<RankedTensorType>(getSource().getType());
+  auto resultType = cast<RankedTensorType>(getResult().getType());
+  if (sourceType.getEncoding() != resultType.getEncoding()) {
+    return emitOpError() << "the source and result of a bitcast should have "
+                            "the same encoding";
+  }
```

Comment:
This is not valid - encodings can (and likely will) differ. Bitcast is the op we use to say "here I have some bytes, reinterpret them as this other type of bytes" - changing shape/data type/encoding is the point of it ;)

```suggestion
```